docs/src/pages/components/typography/typography.md

```diff
@@ -106,6 +106,63 @@ const theme = createMuiTheme({
 });
 ```
 
+## Adding & disabling variants
+
+Apart from the default variants supported, you can add custom variants to use with the `Typography` component, or disable some existing ones. Here is what you need to do:
+
+### Step 1. Update the theme's typography object
+
+```js
+const theme = createMuiTheme({
+  typography: {
+    poster: {
+      color: 'red',
+    },
+    // Disable v3 variant
+    h3: undefined,
+  },
+});
+```
+
+### Step 2. Update the necessary typings if you are using Typescript
+
+**Note:** If you are not using `Typescript` you should skip this step.
+
+You will need to make sure that the typings for the theme's `typography` variants and the `Typogrpahy`'s `variant` prop reflects the new set of variants.
+
+```ts
+declare module '@material-ui/core/styles/createTypography' {
+  interface Typography {
+    poster: React.CSSProperties;
+  }
+
+  // allow configuration using `createMuiTheme`
+  interface TypographyOptions {
+    poster?: React.CSSProperties;
+  }
+}
+
+// Update the Typography's variant prop options
+declare module '@material-ui/core/Typography/Typography' {
+  interface TypographyPropsVariantOverrides {
+    poster: true;
+    h3: false;
+  }
+}
+```
+
+### Step 3. You can now use the new variant
+
+```jsx
+<Typography variant="poster">poster</Typography>;
+{
+  /* This variant is no longer supported! */
+}
+<Typography variant="h3">h3</Typography>;
+```
+
+{{"demo": "pages/components/typography/TypographyCustomVariant.js", "hideToolbar": true}}
```

Comment:
I am hiding this example, as adding the ts module augmentation on the example is being applied on all docs. The instructions are added as code snippets above.

Reply:
I really wish we could isolate the demos regarding module augmentation. We have a couple of places where we use it or should use it. But since these are applied to all files it's quite problematic.

At least so that the type-checking test suite verifies that the module augmentation is working as expected.

Reply:
Agree, this is problematic.. This is the second time I am hitting this issue so far :\